test/common/tmpdir.js

```diff
@@ -1,31 +1,56 @@
 /* eslint-disable node-core/require-common-first, node-core/required-modules */
 'use strict';
 
+const { execSync } = require('child_process');
 const fs = require('fs');
 const path = require('path');
+const { debuglog } = require('util');
 
-function rimrafSync(p) {
-  let st;
-  try {
-    st = fs.lstatSync(p);
-  } catch (e) {
-    if (e.code === 'ENOENT')
-      return;
+const debug = debuglog('test/tmpdir');
+
+function rimrafSync(pathname, { spawn = true } = {}) {
+  const st = (() => {
+    try {
+      return fs.lstatSync(pathname);
+    } catch (e) {
+      if (fs.existsSync(pathname))
+        throw new Error(`Something wonky happened rimrafing ${pathname}`);
+      debug(e);
+    }
+  })();
+
+  // If (!st) then nothing to do.
+  if (!st) {
+    return;
+  }
+
+  // On Windows first try to delegate rmdir to a shell.
+  if (spawn && process.platform === 'win32' && st.isDirectory()) {
+    try {
+      execSync(`rmdir /q /s ${pathname}`, { timout: 1000 });
+    } catch (e) {
+      debug(e);
+    }
   }
 
   try {
-    if (st && st.isDirectory())
-      rmdirSync(p, null);
+    if (st.isDirectory())
+      rmdirSync(pathname, null);
     else
-      fs.unlinkSync(p);
+      fs.unlinkSync(pathname);
   } catch (e) {
-    if (e.code === 'ENOENT')
-      return;
-    if (e.code === 'EPERM')
-      return rmdirSync(p, e);
-    if (e.code !== 'EISDIR')
-      throw e;
-    rmdirSync(p, e);
+    switch (e.code) {
+      case 'ENOENT':
+        return;
+
+      case 'EPERM':
+        return rmdirSync(pathname, e);
+
+      case 'EISDIR':
+        throw e;
```

Comment:
Previously the error was thrown if `e.code !== 'EISDIR'`, is this correct?

Reply:
Nope, it's wrong... I did the `!` didn't register in my mind.
Fixing.